rustls/src/client/client_conn.rs

```diff
@@ -219,34 +217,38 @@ pub struct ClientConfig {
 }
 
 impl ClientConfig {
-    /// Create a builder for a client configuration with the default
-    /// [`CryptoProvider`]: [`crypto::ring::default_provider`] and safe ciphersuite and
-    /// protocol defaults.
+    /// Create a builder for a client configuration with the process-default
+    /// [`CryptoProvider`]: [`CryptoProvider::process_default`] and safe
+    /// protocol version defaults.
     ///
     /// For more information, see the [`ConfigBuilder`] documentation.
-    #[cfg(feature = "ring")]
     pub fn builder() -> ConfigBuilder<Self, WantsVerifier> {
-        // Safety: we know the *ring* provider's ciphersuites are compatible with the safe default protocol versions.
-        Self::builder_with_provider(crate::crypto::ring::default_provider().into())
-            .with_safe_default_protocol_versions()
-            .unwrap()
+        Self::builder_with_protocol_versions(versions::DEFAULT_VERSIONS)
     }
 
-    /// Create a builder for a client configuration with the default
-    /// [`CryptoProvider`]: [`crypto::ring::default_provider`], safe ciphersuite defaults and
+    /// Create a builder for a client configuration with the process-default
+    /// [`CryptoProvider`]: [`CryptoProvider::process_default`] and
     /// the provided protocol versions.
     ///
-    /// Panics if provided an empty slice of supported versions.
+    /// Panics if
+    /// - the supported versions are not compatible with the provider (eg.
+    ///   the combination of ciphersuites supported by the provider and supported
+    ///   versions lead to zero cipher suites being usable),
+    /// - if a `CryptoProvider` cannot be resolved a combination of
+    ///   the crate features and process default.
```

Comment:
I think a word might have been dropped here. Maybe:

```suggestion
    /// - if a `CryptoProvider` cannot be resolved using a combination of
    ///   the crate features and process default.
```